numpy/core/numeric.py

```diff
@@ -2315,15 +2317,26 @@ def array_equal(a1, a2):
     False
     >>> np.array_equal([1, 2], [1, 4])
     False
-
+    >>> a = np.array([1, np.nan])
+    >>> np.array_equal(a, a)
+    False
+    >>> np.array_equal(a, a, equal_nan=True)
+    True
     """
     try:
         a1, a2 = asarray(a1), asarray(a2)
     except Exception:
         return False
     if a1.shape != a2.shape:
         return False
-    return bool(asarray(a1 == a2).all())
+    if not equal_nan:
+        return bool(asarray(a1 == a2).all())
+    # Handling NaN values if equal_nan is True
+    a1nan, a2nan = isnan(a1), isnan(a2)
+    # NaN's occur at different locations
+    if not all(a1nan == a2nan):
```

Comment:
Is `all` imported here, maybe just use (a1nan == n2nan).all() that is also used above. There is an `asarray` call above to work around some scalar issues. I do not thin kit is necessary here though.

Reply:
Good catch! I agree with you: it reads nicer to match the `.all()` used immediately above.